internal/resource/config/oauth/client/oauth_client_common.go

```diff
@@ -228,11 +234,19 @@ func readOauthClientResponseCommon(ctx context.Context, r *client.Client, state
 	state.JwksSettings = jwksSettingsToState
 
 	// state.ExtendedParameters
-	extendedParametersAttrType := map[string]attr.Type{}
-	extendedParametersAttrType["values"] = types.SetType{ElemType: types.StringType}
-	extendedParametersObjAttrType := types.ObjectType{AttrTypes: extendedParametersAttrType}
-	extendedParametersToState, respDiags := types.MapValueFrom(ctx, extendedParametersObjAttrType, r.ExtendedParameters)
-	diags.Append(respDiags...)
+	var extendedParametersToState basetypes.MapValue
+	if r.ExtendedParameters != nil {
+		extendedParametersToState, respDiags = types.MapValueFrom(ctx, extendedParametersObjAttrType, r.ExtendedParameters)
+		diags.Append(respDiags...)
+	} else {
+		if internaltypes.IsDefined(plan.ExtendedParameters) {
+			extendedParametersToState, respDiags = types.MapValueFrom(ctx, extendedParametersObjAttrType, plan.ExtendedParameters)
```

Comment:
I don't think you need to use MapValueFrom here - plan.ExtendedParameters should already be the Map you need.

Reply:
updated to below
`if plan != nil && internaltypes.IsDefined(plan.ExtendedParameters) {
			extendedParametersToState = plan.ExtendedParameters`